src/Analyzer/Passes/RewriteJoinUseNullsPass.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+#include <Analyzer/IQueryTreePass.h>
+
+namespace DB
+{
+/**
+ *  If join_use_nulls = true, then rewrite join nide
+ *  - if it's full join, make all columns from left/right table to nullable;
+ *  - if it's left join, make all columns from right table to nullable;
+ *  - if it's right join, make all columns from left table to nullable;
+ */
+class RewriteJoinUseNullsPass final : public IQueryTreePass
```

Comment:
I suspect that this will not work and it must be implemented in QueryAnalysisPass, because in edge cases, we need to know proper type of expression during query analysis pass.
Example: 
`SELECT cast(lhs.column_1, toTypeName(rhs.column_2)) FROM test_table_1 AS lhs LEFT JOIN test_table_2 AS rhs ON lhs.id = rhs.id`.
Here we want to have proper type of `rhs.column_2`, and `join_use_nulls` must be applied when we resolve `rhs.column_2` identifier.

Reply:
Thank for pointing out this. what about ignoring `join_node_in_resolve_process` here ?
![image](https://github.com/ClickHouse/ClickHouse/assets/2599447/02720104-b8e1-4131-8aa0-e70ec1b636a1)

And add a to nullable projection for the table.